### DISCUSSION TITLE
Recent 20ms delay added to visual waveform cause some songs to render out of bound.

### DISCUSSION THREAD

**Discussion by Trung0246** [ppy/osu#26208] [Q&A]
### Type

Cosmetic

### Bug description

I believe this is the root cause: https://github.com/ppy/osu/pull/26136

Is it possible to move the constant to a separate settings and set that value to 20 if we intend to keep that constant?

Affected songs:
https://osu.ppy.sh/beatmapsets/2047843#osu/4275701
https://osu.ppy.sh/beatmapsets/1312305#osu/2719816
https://osu.ppy.sh/beatmapsets/1846720#osu/3793506
https://osu.ppy.sh/beatmapsets/1017505#osu/2129555

All of these songs have the characteristic of 0ms offset timing.

### Screenshots or videos

<img width="355" alt="Screenshot 2023-12-28 153937" src="https://github.com/ppy/osu/assets/11626920/93de9134-05ed-4d90-808a-f666da2f07d5">
<img width="423" alt="Screenshot 2023-12-28 153843" src="https://github.com/ppy/osu/assets/11626920/b6d9ab23-4552-4e67-8a37-c22c0af2d4ff">
<img width="623" alt="Screenshot 2023-12-28 153621" src="https://github.com/ppy/osu/assets/11626920/878a342a-ca11-4397-96a5-e67f27d09a1a">

### Version

2023.1229.0-lazer

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/13790107/compressed-logs.zip)

**Comment by peppy** [comment]
I believe you are misunderstanding. You NEED to time to the new offset as all players expect this timing.

You cannot use external waveforms to time. osu! beatmaps have an offset and with this change we calculated the offset to be applied to bring all waveforms in line.

**Comment by Trung0246** [comment]
Reposting my thought here from discord:

I think this would probably made things even more confusing, as new players would see the waveform being "off", and hence would see this as a bug (even without using external software).

I think for 0ms song cases like these, the best way would be just add a setting to optionally disable this extra visual delay, and then apply the extra 20ms delay when exporting and saving (maybe for legacy purpose just apply when exporting to the old format).

**Reply by peppy** [reply]
Again, I don't know how to better explain this.

If you listen to the song, the timing is correct when aligned with the waveform. It is all correct. The waveform being "off" here has not changed where you can / cannot place hitobjects. It's just a visual reference.

**Comment by Trung0246** [comment]
Discord discussion thread just in case: https://discord.com/channels/188630481301012481/1189078742555951214

For anyone wants to try if 20ms visual delay does affect anything on their side, try retiming these songs (reposted from discord).

- You can start by removing the old timing point and set the new time point at -20ms per the current latest version as taking account into the visual delay.

The quirks of all these songs is their waveform and sound begins exactly at 0ms if 20ms visual delay did not applied, hence in theory should be the calibration basepoint.

https://osu.ppy.sh/beatmapsets/217935#osu/511046
https://osu.ppy.sh/beatmapsets/1601364#osu/3270303
https://osu.ppy.sh/beatmapsets/2085335#osu/4368604
https://osu.ppy.sh/beatmapsets/1628048#osu/3323777
https://osu.ppy.sh/beatmapsets/2044096#osu/4266258 (more chiptune) 
https://osu.ppy.sh/beatmapsets/1197928#osu/2495113
https://osu.ppy.sh/beatmapsets/534281#osu/1131748
https://osu.ppy.sh/beatmapsets/790841#osu/2085629 (this have very high kick and probably best one since chiptune) 
https://osu.ppy.sh/beatmapsets/1160351#osu/3320181
https://osu.ppy.sh/beatmapsets/1687287#osu/3448347
https://osu.ppy.sh/beatmapsets/2094243#osu/4390992
https://osu.ppy.sh/beatmapsets/1237886#osu/2585251
https://osu.ppy.sh/beatmapsets/1547726#osu/3162870 (chiptune of course)
